From 5b6d6706bc7bef32ac9a367d92f8a19dcdbb89c4 Mon Sep 17 00:00:00 2001
From: crazy <crazy@frugalware.org>
Date: Sat, 2 Jan 2016 03:30:36 +0100
Subject: [PATCH] repoman * added check for broken paths :   /usr/usr  
 /usr/share/pkgconfig   /usr/share/perl5 * and refuse to upload the package

---
 repoman | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/repoman b/repoman
index 634b413..a29ada9 100755
--- a/repoman
+++ b/repoman
@@ -403,6 +403,25 @@ upload()
 			echo "Please fix your package."
 			exit 1
 		fi
+		if echo $src | grep -q fpm$ && pacman-g2 -Ql -p $src | grep -q 'usr/usr'; then
+                        echo "$src has broken /usr DIR , files are installed"
+                        echo "in /usr/usr , refusing to upload!"
+                        echo "Please fix your package."
+                        exit 1
+		fi
+		if echo $src | grep -q fpm$ && pacman-g2 -Ql -p $src | grep -q 'usr/share/pkgconfig'; then
+                        echo "$src has invalid pkgconfig path , files are installed"
+                        echo "in /usr/share/pkgconfig , should be /usr/lib/pkgconfig, refusing to upload!"
+                        echo "Please fix your package."
+                        exit 1
+		fi
+		if echo $src | grep -q fpm$ && pacman-g2 -Ql -p $src | grep -q 'usr/share/perl5'; then
+                        echo "$src has invalid perl path , files are installed"
+                        echo "in /usr/share/perl5 , refusing to upload!"
+                        echo "Please fix your package."
+                        exit 1
+		fi
+
 		name=`basename $src`
 		msg "Requesting a temporary file"
 		tmp=`_ssh "mktemp"`
-- 
2.6.4

